### PR TITLE
Backend.put_archive(): make `files` an optional argument

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes https://github.com/audeering/audbackend/issues/88

image

### CODE REVIEW

File: audbackend/core/backend.py [hunk 1]
```diff
@@ -489,31 +493,23 @@ def put_archive(
...
 
-        if not os.path.exists(src_root):
-            utils.raise_file_not_found_error(src_root)
-
-        files = audeer.to_list(files)
-
-        for file in files:
-            path = os.path.join(src_root, file)
-            if not os.path.exists(path):
-                utils.raise_file_not_found_error(path)
```
Comment (hagenw) [inline audbackend/core/backend.py:515]: I guess you were able to remove this code as those errors are now raised by `audeer.create_archive()`?

Comment (frankenjoe) [reply audbackend/core/backend.py:515]: yes

File: tests/test_backend.py [hunk 2]
```diff
@@ -9,70 +9,85 @@
...
-        (
+        (  # all files
             ['file.ext', 'dir/to/file.ext'],
-            'not-empty.zip',
-            'group',
-            '2.0.0',
+            '/archive.zip',
+            None,
             'tmp',
+            ['dir/to/file.ext', 'file.ext'],
```
Comment (hagenw) [inline tests/test_backend.py:65]: Could you add a test here with `files=['file.ext', 'dir/to/file.ext']`. If I remember correctly, we should then have `expected=['file.ext', 'dir/to/file.ext']`.

Comment (frankenjoe) [reply tests/test_backend.py:72]: done

Comment (hagenw) [reply tests/test_backend.py:72]: Not directly you added a test for:
```
files=['dir/to/file.ext', 'file.ext']
expected=['dir/to/file.ext', 'file.ext']
```
but I asked for
```
files=['file.ext', 'dir/to/file.ext']
expected=['file.ext', 'dir/to/file.ext']
```
to see if the order of the provided files has an influence on expected paths.
So maybe just add another test with the reversed list.

Comment (frankenjoe) [reply tests/test_backend.py:72]: Ah yes, my bad. I fixed it.

**Pull request merged**